tests/Unit/TestCreation.hpp

```diff
@@ -80,8 +90,13 @@ std::unique_ptr<BaseClass> test_factory_creation(
 }
 
 /// \ingroup TestingFrameworkGroup
-/// Construct an enum from a given string. The string must not contain the tag
-/// name.
+/// Construct an enum from a given string.
+///
+/// The string must not contain the
+/// name of the option (which is the name of the struct by default if no
+/// `name()` function is present). For example, to create a struct with a
+/// `SizeT` named option the following would be used:
+/// \snippet Test_TestCreation.cpp size_t_argument
```

Comment:
This is enum creation; why not demonstrate the parsing with an enum? e.g. line 387ish of the test.